client.go

```diff
@@ -357,3 +380,153 @@ func prepareURL(url *url.URL, path string) (*url.URL, error) {
 
 	return u, nil
 }
+
+// prepareQueryParameters Allows for easy preparation of query string
+func (cl *Client) prepareQueryParameters(params interface{}) (queryString string, err error) {
+	v, err := query.Values(params)
+
+	// If we do have account switch key - we will add it and toggle ASK back to disabled
+	if cl.accountSwitchEnabled == true {
+		v.Add("accountSwitchKey", cl.accountSwitchKey)
+		cl.accountSwitchEnabled = false
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	return v.Encode(), nil
+}
+
+// makeAPIRequest creates an HTTP request that can be sent to Akamai APIs. It will handle security headers and signinig of the request.
+//
+func (cl *Client) makeAPIRequest(method, path string, queryParams, structResponse, structRequest interface{}, headers map[string]string) (*ClientResponse, error) {
+
+	log.Debug("[NewRequest]::Prepare URL for http request")
+	targetURL, _ := prepareURL(cl.baseURL, path)
```

Comment:
Do not ignore error, handle it

Reply:
Fixed https://github.com/apiheat/go-edgegrid/pull/21/commits/97f4a8d8dc4cd13cf194c3a300ccff67fd8d7045